### PR TITLE
Backport stack frame changes

### PR DESCRIPTION
**Pull request by petertdavies**
### What was wrong?

The change to the stack frames had only been applied to the London hardfork.

### How was it fixed?

Port it to all the other forks.

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: src/ethereum/berlin/vm/__init__.py [hunk 1]
```diff
@@ -14,16 +14,19 @@
...
 from ethereum.crypto.hash import Hash32
 
 from ..eth_types import Address, Log
-from ..state import State
+from ..state import State, account_exists_and_is_empty
+from ..utils.address import to_address
 
 __all__ = ("Environment", "Evm", "Message")
 
+RIPEMD160_ADDRESS = to_address(Uint(3))
```
Comment (SamWilsn) [inline src/ethereum/berlin/vm/__init__.py:28]: I think you can do this now:

```python
from .precompiled_contracts.mapping import RIPEMD160_ADDRESS
```

File: src/ethereum/berlin/vm/__init__.py [hunk 2]
```diff
@@ -81,10 +84,62 @@ class Evm:
...
+    Incorporate the state of an unsuccessful `child_evm` into the parent `evm`.
+
+    Parameters
+    ----------
+    evm :
+        The parent `EVM`.
+    child_evm :
+        The child evm to incorporate.
+    """
+    # In block 2675119, the empty account at 0x2 (the RIPEMD160 precompile) was
```
Comment (SamWilsn) [inline src/ethereum/berlin/vm/__init__.py:131]: Isn't the RIPEMD160 precompile at 0x3?

File: src/ethereum/berlin/vm/__init__.py (hunk 1 above)
Comment (petertdavies) [reply src/ethereum/berlin/vm/__init__.py]: Unfortunately this fails due to a circular import.

Comment (SamWilsn) [reply src/ethereum/berlin/vm/__init__.py:28]: Hm, that's a problem then. I'd rather not regress on #643. Could we move the `*_ADDRESS` constants to `__init__.py` maybe?

**Review by gurukamath — APPROVED** [review]

**Pull request merged**